pkg/ring/lifecycler.go

```diff
@@ -778,6 +781,16 @@ func (i *Lifecycler) SetFlushOnShutdown(flushOnShutdown bool) {
 	i.flushOnShutdown.Store(flushOnShutdown)
 }
 
+// SkipUnregister returns if unregistering is skipped on shutdown.
+func (i *Lifecycler) UnregisterFromRing() bool {
```

Comment:
**suggestion(non-blocking,if-minor):** I would rename this function to parallel `SetUnregisterFromRing`. In my opinion `UnregisterFromRing()` connotes that it will actually perform the operation to unregister from the ring, not return the config value.

```suggestion
func (i *Lifecycler) GetUnregisterFromRing() bool {
```

Reply:
I changed it to `Should` which read nicer to me than `Get`.